### ISSUE TITLE
External lexers and function list and/or autocompletion functionality

### ISSUE THREAD

**Issue by Ekopalypse** [notepad-plus-plus/notepad-plus-plus#9519]
External lexers coming via a plugin can currently not use existing functionalities like
Function-List and/or Completion via API.xml because, as far as I understand the e.g. function list code
then every ID > L_EXTERNAL is considered to be an UDL which in case of a plugin lexer is not the case.

Question: Is this desired, since it is assumed that a plugin lexer provides these functionalities itself?

**Comment by donho** [comment]
@Ekopalypse 
So if I do understand your question, you want to know how to use function list and autocompletion with external lexers, right?

**Comment by Ekopalypse** [comment]
@donho 
Yes, or if it is assumed that external lexers implement these functionalities themselves.
